### DISCUSSION TITLE
Does AssetLoader::CreateAssetFromJson has to work in the same thread that create filament::Engine?

### DISCUSSION THREAD

**Discussion by JackKaiXing** [google/filament#3995] [Q&A] [answered]
thank.

`AssetLoader::CreateAssetFromJson` does not return immediately, for one of our model, it costs like 8ms.

If we seperate `AssetLoader::CreateAssetFromJson` in new thread, it crashs at `MaterialProvider::CreateMaterialInstance`.

Does it possible that we create all the needed `materialInstance` in advance?

**Comment by prideout** [comment, marked as answer]
Yes, it needs to be called from the same thread that was used to create Engine.

When you create an AssetLoader, you pass in a MaterialProvider.  I think your startup time will be much faster if you pass in a provider that was created using `createUbershaderLoader` rather than `createMaterialGenerator`.
